include/xtensor/xbuffer_adaptor.hpp

```diff
@@ -816,6 +853,15 @@ namespace xt
             swap(m_size, rhs.m_size);
             swap(m_destruct, rhs.m_destruct);
         }
+
+        template <class CP, class D>
+        template <class P, class DT>
+        void xbuffer_smart_pointer<CP, D>::replace(P&& data, size_type size, DT&& destruct) noexcept
+        {
+            p_data = data;
```

Comment:
`data` should be forwarded (it you use a `unique_ptr`, this code does not compile).